tx/textual/valuerenderer/int.go

```diff
@@ -26,5 +31,62 @@ func (vr intValueRenderer) Format(_ context.Context, v protoreflect.Value) ([]Sc
 }
 
 func (vr intValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	panic("implement me")
+	if len(screens) != 1 {
+		return nilValue, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	parsedInt, err := parseInt(screens[0].Text)
+	if err != nil {
+		return nilValue, err
+	}
+
+	switch vr.fd.Kind() {
+	case protoreflect.Uint32Kind:
+		value, err := strconv.ParseUint(parsedInt, 10, 32)
+		if err != nil {
+			return nilValue, err
+		}
+		return protoreflect.ValueOfUint32(uint32(value)), nil //nolint:gosec
+
+	case protoreflect.Uint64Kind:
+		value, err := strconv.ParseUint(parsedInt, 10, 64)
+		if err != nil {
+			return nilValue, err
+		}
+		return protoreflect.ValueOfUint64(value), nil
+
+	case protoreflect.Int32Kind:
+		value, err := strconv.ParseInt(parsedInt, 10, 32)
+		if err != nil {
+			return nilValue, err
+		}
+		return protoreflect.ValueOfInt32(int32(value)), nil //nolint:gosec
+
+	case protoreflect.Int64Kind:
+		value, err := strconv.ParseInt(parsedInt, 10, 64)
+		if err != nil {
+			return nilValue, err
+		}
+		return protoreflect.ValueOfInt64(value), nil
+
+	case protoreflect.StringKind:
+		return protoreflect.ValueOfString(parsedInt), nil
+
+	default:
+		return nilValue, fmt.Errorf("parsing integers into a %s field is not supported", vr.fd.Kind())
+	}
+}
+
+// parseInt parses a value-rendered string into an integer
+func parseInt(v string) (string, error) {
+	sign := ""
+	if v[0] == '-' {
+		sign = "-"
+		v = v[1:]
+	}
+
+	// remove the 1000 separators (ex: 1'000'000 -> 1000000)
+	v = strings.Replace(v, "'", "", -1)
+
+	return sign + v, nil
```

Comment:
Oops unfortunately this code does nothing to validate that we are passing in integers. We need to do more work in here after the replacement of the thousands separator like this
```go
func hasOnlyDigits(v string) bool {
   if v == "" {
       return false
   }
   for _, r := range v {
         if r < '0' || r > '9' {
             return false
         }
   }
   return true
}

...
v = strings.ReplaceAll(v, thousandsSeparator, "")
if !hasOnlyDigits(v) {
     return "", fmt.Errorf("expecting only digits 0-9 but got %q", v)
}
return sign+v, nil
```